lib/ansible/modules/database/mongodb/mongodb_user.py

```diff
@@ -400,16 +400,13 @@ def main():
 
         client = MongoClient(**connection_params)
 
-        # NOTE: this check must be done ASAP.
-        # We doesn't need to be authenticated.
-        check_compatibility(module, client)
-
         if login_user is None and login_password is None:
             mongocnf_creds = load_mongocnf()
             if mongocnf_creds is not False:
                 login_user = mongocnf_creds['user']
                 login_password = mongocnf_creds['password']
-        elif login_password is None or login_user is None:
+
+        if login_password is None or login_user is None:
```

Comment:
This is wrong. Now if you don't supply login credentials at all, it will error out. So goodbye localhost exception to create the first user.
The "elif" branch must not be deleted.

Reply:
Well, but seems check_compatibility cannot be used even with the localhost exception...so no solution.